### PR TITLE
Added entity framework UoW sample

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Requires new API to be added to NHibernate persistence: https://github.com/Particular/NServiceBus.NHibernate/pull/266

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@HEskandari would this address the problem from the case you're dealing with?

@andreasohlund @timbussmann I think this is good enough. Please take a look at the actual UoW implementation classes and tell me if this is not too hacky

File: samples/entity-framework/Core_6/Receiver/CreateOrderHandler.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class CreateOrderHandler :
+    IHandleMessages<OrderSubmitted>
+{
+    static ILog log = LogManager.GetLogger<CreateOrderHandler>();
+
+    public Task Handle(OrderSubmitted message, IMessageHandlerContext context)
+    {
+        log.Info($"Order {message.OrderId} worth {message.Value} submitted");
+
+        #region StoreOrde
```
Comment (andreasohlund) [inline samples/entity-framework/Core_6/Receiver/CreateOrderHandler.cs]: Typo

File: samples/entity-framework/Core_6/Receiver/UnitOfWork/EntityFrameworkUnitOfWork.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        var storageSession = context.SynchronizedStorageSession;
+
+        var dbConnection = storageSession.Session().Connection;
+        Context = new ReceiverDataContext(dbConnection);
+
+        //Don't use transaction because connection is enlisted in the TransactionScope
+        Context.Database.UseTransaction(null);
+
+        //Call SaveChanges before completing storage session
+        storageSession.RegisterCommitHook(() => Context.SaveChangesAsync());
```
Comment (andreasohlund) [inline samples/entity-framework/Core_6/Receiver/UnitOfWork/EntityFrameworkUnitOfWork.cs]: Is `RegisterCommitHook` something you've added to NH?

Comment (SzymonPobiega) [inline samples/entity-framework/Core_6/Receiver/UnitOfWork/EntityFrameworkUnitOfWork.cs]: Yeah. Well, I plan to add ;-)

Comment (andreasohlund) [inline samples/entity-framework/Core_6/Receiver/UnitOfWork/EntityFrameworkUnitOfWork.cs]: Pseudo code ftw! :)

**Comment by HEskandari** [comment]
@SzymonPobiega at first look this does seem to address the issue. Should I confirm with the customer?

**Comment by SzymonPobiega** [comment]
@HEskandari it would be awesome.

**Comment by weralabaj** [comment]
@HEskandari @SzymonPobiega updates?

Is that waiting for new nhibernate release?

**Comment by HEskandari** [comment]
I have asked the customer if this addresses the issue. Waiting for their feedback. I'll ping the once again and update here as soon as I hear.

**Comment by SzymonPobiega** [comment]
@weralabaj @SimonCropp would you say it makes sense to have such a sample regardless of the fact if it solves this particular customer's problem?

**Comment by weralabaj** [comment]
It might, but could we add some intro section explaining why should you care and what problem it solves? just a couple of sentences will be fine.

**Comment by SimonCropp** [comment]
@SzymonPobiega i dont mind either way.

**Comment by SimonCropp** [comment]
@SzymonPobiega ping

**Comment by SzymonPobiega** [comment]
@SimonCropp I remember about it. I still wait for NH to be released so that I can progress with it.

**Comment by SzymonPobiega** [comment]
I've cleaned up the code and it is ready for a proper review @Particular/docs-maintainers

**Comment by SzymonPobiega** [comment]
It's green. @Particular/docs-maintainers OK to merge?

**Comment by ramonsmits** [comment]
@SzymonPobiega I just tried this sample and it doesn't work at runtime and getting a `Invalid object name 'dbo.Shipments'`
`

**Comment by ramonsmits** [comment]
@SzymonPobiega That was strange, it seems my DB already had an 'Orders' table which prevented EF to do a migration or something. After dropping that table the Orders and Shipments tables got created.

**Pull request merged**